system/jlib/jfile.cpp

```diff
@@ -7643,3 +7643,110 @@ IAPICopyClient * createApiCopyClient(IStorageApiInfo * source, IStorageApiInfo *
+{
+    Owned<IFileIO> io;
+    offset_t fileSize = 0;
+    size32_t blockSize = 0;
+public:
+    CBlockedFileIO(IFileIO *_io, size32_t _blockSize) : io(_io), blockSize(_blockSize)
+    {
+        fileSize = io->size();
+    }
+    ~CBlockedFileIO()
+    {
+        blockedFileIOLastReadCaller = nullptr;
+    }
+    static bool onThreadTerm(bool isPooled)
+    {
+        free(blockedFileIOBuffer);
+        blockedFileIOBuffer = nullptr;
+        return false;
+    }
+    virtual size32_t read(offset_t pos, size32_t len, void *data) override
+    {
+        if (len > blockSize || pos+len > fileSize || len==0)
+            return io->read(pos, len, data);
+        if (!blockedFileIOBuffer)
+        {
+            blockedFileIOBuffer = malloc(blockSize);
+            addThreadTermFunc(onThreadTerm);
+        }
+        size32_t totalCopied = 0;
+        byte *dest = (byte *) data;
+        while (len)
+        {
+            offset_t readPos = (pos / blockSize) * blockSize;
+            size32_t readLen;
+            if (readPos + blockSize <= fileSize)
+                readLen = blockSize;
+            else
+                readLen = fileSize - readPos;
+            if ((readPos != blockedFileIOLastReadPos) || (this != blockedFileIOLastReadCaller)) // NB: if readLen < blockSize then last block, won't ever read more for that block.
+            {
+                size32_t read = io->read(readPos, readLen, blockedFileIOBuffer);
+                assertex(read == readLen);
+                blockedFileIOLastReadPos = readPos;
+                blockedFileIOLastReadCaller = this;
+            }
+            size32_t copyNow;
+            if (pos+len <= readPos+readLen)
+                copyNow = len;
+            else
+                copyNow = readPos+readLen-pos;
+            memcpy(dest, ((byte *)blockedFileIOBuffer) + pos-readPos, copyNow);
+            len -= copyNow;
+            pos += copyNow;
+            dest += copyNow;
+            totalCopied += copyNow;
+        }
+        return totalCopied;
+    }
+    virtual offset_t size() override { return fileSize; }
+    virtual size32_t write(offset_t pos, size32_t len, const void * data) override { throwUnexpected(); }
+    virtual offset_t appendFile(IFile *file, offset_t pos=0, offset_t len=(offset_t)-1) override { throwUnexpected(); }
+    virtual void setSize(offset_t size) override { throwUnexpected(); }
+    virtual void flush() override { throwUnexpected(); }
+    virtual void close() override { io->close(); }
+    virtual unsigned __int64 getStatistic(StatisticKind kind) override { return io->getStatistic(kind); }
+};
+
+extern IFileIO *createBlockedIO(IFileIO *base, size32_t blockSize)
+{
+    PROGLOG("createBlockedIO(%u)", (unsigned)blockSize);
+    return new CBlockedFileIO(base, blockSize);
+}
+
+static constexpr size32_t defaultBlockedIndexIOKB = 1024;
+size32_t getBlockedIndexIOSize(const char *planeName, int configUseBlockedIndexIO, size32_t configBlockedIndeIOSize)
```

Comment:
Spelling: configBlockedIndeIOSize should be configBlockedIndexIOSize

Reply:
will correct